src/app/modules/group/components/joined-group-list/joined-group-list.component.ts

```diff
@@ -0,0 +1,58 @@
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { SortEvent } from 'primeng/api';
+import { merge, of, Subject } from 'rxjs';
+import { map, switchMap } from 'rxjs/operators';
+import { JoinedGroup, JoinedGroupsService } from 'src/app/core/http-services/joined-groups.service';
+import { fetchingState, isReady, readyState } from 'src/app/shared/helpers/state';
+
+@Component({
+  selector: 'alg-joined-group-list',
+  templateUrl: './joined-group-list.component.html',
+  styleUrls: [ './joined-group-list.component.scss' ]
+})
+export class JoinedGroupListComponent implements OnDestroy, OnInit {
+
+  state: 'error' | 'ready' | 'fetching' = 'fetching';
+  currentSort: string[] = [];
+
+  data: JoinedGroup[] = [];
+
+  private dataFetching = new Subject<{ sort: string[] }>();
```

Comment:
Other similar components use `reloadData` functions and `unsubscribe`. I am not sure it was worth changing this here... but ok, leave it.